text/stacks/0003-stack-descriptor.md

```diff
@@ -0,0 +1,88 @@
+# Stack Descriptor and Tooling
+
+## Summary
+
+The stack-building codebase,
+[`create-stack`](https://github.com/paketo-buildpacks/stacks/tree/main/create-stack)
+should be moved into the `jam` tooling CLI, and allow users to configure their
+stack build via a common configuration file.
+
+## Motivation
+
+Today, `create-stack` is highly coupled to the `bionic` stacks we currently
+support. This makes it difficult to support alternative stacks and platforms or
+to allow the codebase to be reused by external parties for developing their own
+stack.
+
+## Detailed Explanation
+
+The `create-stack` codebase will be refactored to allow users to build stacks
+using a configuration file with the following format:
+
+```toml
+id = "<stack id>"                # required
+homepage = "<homepage uri>"      # optional
+maintainer = "<maintainer name>" # optional
+
+platforms = ["linux/amd64", "linux/arm64", "<other-platform>"] # optional; default = ["linux/amd64"]; choices defined in https://docs.docker.com/desktop/multi-arch/
```

Comment:
Aren't the platform(s) are dictated by the platform(s) base image(s) from which we construct the stack?

Maybe we should:
1. require that the dockerfile accept the base image as a build arg
2. require that provide a reference to the base image in this file
3. If the reference is a single image, inherit the platform
4. If the reference is a multi-arch image, build multiple times times and create a multi-arch image from the result

Reply:
I outlined a bit how I expect the toolchain to handle multi-arch images. The outputs will basically always be formatted as an image index, but may only contain images for a single platform.